dis_snek/models/snek/listener.py

```diff
@@ -32,7 +32,14 @@ def wrapper(coro: Coroutine) -> "Listener":
             name = event_name
 
             if name is MISSING:
-                name = coro.__name__
+                if not coro.__name__.startswith("_"):
```

Comment:
Why exactly is this necessary?

Reply:
This would prevent users from using private methods for listeners, which is a very common use case. 

Imo the order of operations ought to be 
`check for passed parameter dictating event` -> `check coroutine name` -> `check argument type hints`

This pr, assuming im reading it correctly would reverse the latter 2
`check for passed parameter dictating event` -> `check argument type hints` -> `check coroutine name`
imo this is not desirable and the new functionality wouldn't even work for private methods.